pkg/event/event_suite_test.go

```diff
@@ -0,0 +1,140 @@
+package event_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	k8sV1 "k8s.io/api/core/v1"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/event"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestRegistryMethods(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Event Registry suite")
+}
+
+type TestEvent struct {
+	Timestamp            int64
+	Name                 string
+	HandlerName          string
+	HandlerNetworkPlugin string
+	Parameter            interface{}
+}
+
+type TestHandler struct {
+	event.HandlerBase
+	Name          string
+	NetworkPlugin string
+	FailOnEvent   error
+	EventList     []TestEvent
+	Initialized   bool
+}
+
+func NewTestHandler(name, networkPlugin string) *TestHandler {
+	return &TestHandler{
+		Name:          name,
+		NetworkPlugin: networkPlugin,
+		EventList:     []TestEvent{},
+		FailOnEvent:   fmt.Errorf("Test handler not initialized"),
+		Initialized:   false,
+	}
+}
+
+func (t *TestHandler) CleanupEvents() {
+	t.EventList = []TestEvent{}
+}
+
+func (t *TestHandler) addEvent(eventName string, param interface{}) error {
+	ev := TestEvent{
+		Timestamp:            time.Now().UnixNano(),
+		Name:                 eventName,
+		HandlerNetworkPlugin: t.NetworkPlugin,
+		HandlerName:          t.Name,
+		Parameter:            param,
+	}
+
+	t.EventList = append(t.EventList, ev)
+
+	return t.FailOnEvent
+}
+func (t *TestHandler) Init() error {
+	t.FailOnEvent = nil
+	t.Initialized = true
+
+	return nil
+}
+
+func (t *TestHandler) GetName() string {
+	return t.Name
+}
+
+func (t *TestHandler) GetNetworkPlugin() string {
+	return t.NetworkPlugin
+}
+
+const evTransitionToNonGateway = "TransitionToNonGateway"
+const evTransitionToGateway = "TransitionToGateway"
+const evLocalEndpointCreated = "LocalEndpointCreated"
+const evLocalEndpointUpdated = "LocalEndpointUpdated"
+const evLocalEndpointRemoved = "LocalEndpointRemoved"
+const evRemoteEndpointCreated = "RemoteEndpointCreated"
+const evRemoteEndpointUpdated = "RemoteEndpointUpdated"
+const evRemoteEndpointRemoved = "RemoteEndpointRemoved"
+const evNodeCreated = "NodeCreated"
+const evNodeUpdated = "NodeUpdated"
+const evNodeRemoved = "NodeRemoved"
+const evStop = "Stop"
```

Comment:
Can we move this to handler.go so that we can reuse them even in registry.go

Reply:
Hmm, I tend to prefer separating the constants of test & real implementation, so we double check that we are writting them all correct. Also if somebody inadvertedly changes the const on implementation it will fail the test.

Reply:
Okay